gapic-common/lib/gapic/routing_headers/headers_extractor.rb

```diff
@@ -0,0 +1,80 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "gapic/routing_headers/header_binding"
+
+module Gapic
+  module RoutingHeaders
+    class HeadersExtractor
```

Comment:
Unless this class is `@private` we need documentation for the class and its public methods.

Reply:
Will be @private